### PR TITLE
binlog: use stream to pull binlogs

### PR DESCRIPTION
**Pull request by GregoryIan**
use stream to pull binlogs and add notify rpc interface into cistern

### CODE REVIEW

**Comment by GregoryIan** [comment]
@iamxy @tiancaiamao PTAL

File: proto/binlog/cistern.proto [hunk 1]
```diff
@@ -18,6 +18,9 @@ service Cistern {
 
     // GetLatestCommitTS returns the Last binlog commitTS of the TiDB
     rpc GetLatestCommitTS(GetLatestCommitTSReq) returns (GetLatestCommitTSResp) {}
+
+    // Notify notifies a new pump is comming
+    rpc Notify(NotifyReq) returns (NotifyResp) {}
```
Comment (iamxy) [inline proto/binlog/cistern.proto]: any better name or more detailed comment ?

**Comment by iamxy** [comment]
LGTM

File: proto/binlog/cistern.proto [hunk 2]
```diff
@@ -18,6 +18,10 @@ service Cistern {
 
     // GetLatestCommitTS returns the Last binlog commitTS of the TiDB
     rpc GetLatestCommitTS(GetLatestCommitTSReq) returns (GetLatestCommitTSResp) {}
+
+    // Notify notifies all living cisterns that a new pump is comming
```
Comment (iamxy) [inline proto/binlog/cistern.proto]: s/comming/coming

File: proto/binlog/cistern.proto [hunk 3]
```diff
@@ -60,3 +64,9 @@ message GetLatestCommitTSResp {
     // isSynced specifies whether the all binlogs are consumed from pump
     bool isSynced = 2;
 }
+
+message NotifyReq {
```
Comment (iamxy) [inline proto/binlog/cistern.proto]: can this empty req be removed ?

File: proto/binlog/pump.proto [hunk 4]
```diff
@@ -14,8 +14,8 @@ service Pump {
     // A response with an empty errmsg is returned if the binlog is written successfully.
     rpc WriteBinlog(WriteBinlogReq) returns (WriteBinlogResp) {}
 
-    // Obtains a batch of binlog from a given location.
-    rpc PullBinlogs(PullBinlogReq) returns (PullBinlogResp) {}
+    // Sends binlogs in the streaming way from a given location.
```
Comment (iamxy) [inline proto/binlog/pump.proto]: Sends binlog stream from a given location

File: proto/binlog/cistern.proto (hunk 3 above)
Comment (GregoryIan) [inline proto/binlog/cistern.proto]: the protobuf don't support service message without argument, and maybe we would extend our message in the future.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**